### PR TITLE
Sidebar: Add Reader sidebar

### PR DESCRIPTION
**Pull request by kean**
This PR adds the initial infrastructure for managing the Reader sidebar and navigation. This version no longer relies on the `ReaderTab*` infrastructure that is only applicable for the iPhone.

Screenshot 2024-09-03 at 3 04 24 PM

## To Test

- Reader displays all the primary tabs
- You can switch between any of the tabs and the selection is updated
- The tabs are currently displayed with the standard system navigation bars and no "pill" is present

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabItemsStore.swift [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import WordPressFlux
 
-protocol ItemsStore: Observable {
+protocol ReaderTabItemsStoreProtocol: Observable {
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabItemsStore.swift:3]: Renaming to eliminate the name that's too generic.

File: WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabView.swift [hunk 2]
```diff
@@ -132,38 +130,21 @@ extension ReaderTabView {
...
-                buttonContainer.topAnchor.constraint(equalTo: safeTopAnchor),
-                buttonContainer.leadingAnchor.constraint(equalTo: safeLeadingAnchor),
-                buttonContainer.trailingAnchor.constraint(equalTo: safeTrailingAnchor),
-                mainStackViewTopAnchor,
-                mainStackView.trailingAnchor.constraint(equalTo: safeTrailingAnchor),
-                mainStackView.leadingAnchor.constraint(equalTo: safeLeadingAnchor),
-                mainStackView.bottomAnchor.constraint(equalTo: safeBottomAnchor),
-            ])
-        }
+        mainStackViewTopAnchor = mainStackView.topAnchor.constraint(equalTo: buttonContainer.bottomAnchor)
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabView.swift:133]: This code used to change the "pill" layout, but we are no longer going too display it or use this `ReaderTabView` at all.

File: WordPress/Classes/ViewRelated/Reader/Tab Navigation/WPTabBarController+ReaderTabNavigation.swift [hunk 3]
```diff
@@ -41,16 +41,6 @@ extension WPTabBarController {
         return viewModel
     }
 
-    private func makeReaderContentViewController(with content: ReaderContent) -> ReaderContentViewController {
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Reader/Tab Navigation/WPTabBarController+ReaderTabNavigation.swift:44]: It was unused code.

**Review by crazytonyli — APPROVED** [review]

**Pull request merged**